test/lib/ansible_test/_data/requirements/sanity.pylint.in

```diff
@@ -1,2 +1,3 @@
 pylint == 2.15.5  # currently vetted version
+pytest  # needed for ansible_pytest_unit_test_module_mocks.py
```

Comment:
Why is pytest being added to the pylint requirements?

Also, I'm still not convinced we should be including this in ansible, but we can set that aside for now.

Reply:
> Why is pytest being added to the pylint requirements?

@mattclay I think it was necessary because the module containing the fixtures' implementation imports pytest and is being linted by pylint. Pylint tends to follow imports and reports an error when it can't find pytest.

> Also, I'm still not convinced we should be including this in ansible, but we can set that aside for now.

What about getting rid of a variety of implementations in core itself? Would we be open to installing a community collection in our own tests to access the unified patching implementation?
And since it's patching the internal implementation details / private API (as in `ansible.module_utils.basic._ANSIBLE_ARGS`, for example), it seems reasonable that it should remain coupled with the `AnsibleModule` implementation.
I think it belongs in core for the same reason `ansible-test` hasn't moved out.

Reply:
@webknjaz I agree that depending on a private API (`ansible.module_utils.basic._ANSIBLE_ARGS`) isn't ideal, so I've been thinking about how we could support this kind of functionality outside of core. This, along with installing the collection loader are two pain points for anyone trying to write tests or testing tools outside of core -- since neither is exposed as a public or stable API.

One solution to this would be to define a minimal public test-specific API for these capabilities, so they could be used to support development of testing tools outside of core. For example, `pytest` plugin(s) could be created that would leverage these APIs -- although their use wouldn't be limited to `pytest`.

As for consolidating implementations in core's own tests, I think that's a really good idea. For now I'd just expose the fixtures within core's own tests instead of hosting them as a pytest plugin within ansible-test.